dbschema/seeds/008.projects.ts

```diff
@@ -0,0 +1,142 @@
+    step: 'EarlyConversations',
+    mouStart: '2020-01-01',
+    mouEnd: '2023-12-30',
+    estimatedSubmission: '2023-12-01',
+    financialReportPeriod: 'Monthly',
+  },
+  {
+    name: 'Arnor Lake',
+    step: 'FinalizingProposal',
+    mouStart: '2015-01-01',
+    mouEnd: '2018-12-30',
+    estimatedSubmission: '2022-12-01',
+    financialReportPeriod: 'Monthly',
+  },
+  {
+    name: 'Lothlorien',
+    step: 'Active',
+    mouStart: '2020-01-01',
+    mouEnd: '2025-05-30',
+    primaryLocation: 'Greece',
+    financialReportPeriod: 'Quarterly',
+  },
+  {
+    name: 'Emyn Muil',
+    step: 'Active',
+    mouStart: '2019-03-01',
+    mouEnd: '2022-10-30',
+    primaryLocation: 'India',
+    financialReportPeriod: 'Quarterly',
+  },
+  {
+    name: 'South Downs',
+    step: 'FinalizingCompletion',
+    mouStart: '2020-01-01',
+    mouEnd: '2023-12-30',
+    estimatedSubmission: '2023-09-01',
+    primaryLocation: 'United States',
+    financialReportPeriod: 'Monthly',
+  },
+];
+
+const internships: Input[] = [
+  {
+    name: 'Glorfindel - Exegetical Facilitator',
+    step: 'DiscussingChangeToPlan',
+    mouStart: '2021-07-01',
+    mouEnd: '2024-12-30',
+    primaryLocation: 'Sweden',
+    estimatedSubmission: '2024-10-01',
+    financialReportPeriod: 'Monthly',
+  },
+  {
+    name: 'Arwen Evenstar Intern',
+    step: 'PendingConceptApproval',
+    mouStart: '2022-10-01',
+    mouEnd: '2025-12-30',
+    financialReportPeriod: 'Quarterly',
+  },
+  {
+    name: 'Eomer of Rohan Intern',
+    step: 'Active',
+    mouStart: '2022-02-01',
+    mouEnd: '2026-06-30',
+    primaryLocation: 'Egypt',
+    financialReportPeriod: 'Monthly',
+  },
+  {
+    name: 'Cohort of the Ents',
+    step: 'PendingFinancialEndorsement',
+    mouStart: '2022-02-01',
+    mouEnd: '2026-06-30',
+    financialReportPeriod: 'Quarterly',
+  },
+  {
+    name: 'Barliman Butterbur Intern',
+    step: 'OnHoldFinanceConfirmation',
+    mouStart: '2018-02-01',
+    mouEnd: '2022-07-30',
+    primaryLocation: 'Canada',
+    financialReportPeriod: 'Monthly',
+  },
+];
+
+interface Input {
+  name: string;
+  step: (typeof e.Project.Step)['__tstype__'];
+  mouStart: string;
+  mouEnd: string;
+  financialReportPeriod: (typeof e.ReportPeriod)['__tstype__'];
+  primaryLocation?: string;
+  estimatedSubmission?: string;
+}
+
+export default (async function ({ e, db, print }) {
+  const existing = new Set(await e.select(e.Project).name.run(db));
+
+  const projectSeeds = entries({
+    Translation: translations,
+    Internship: internships,
+  }).flatMap(([type, list]) => list.map((item) => ({ type, ...item })));
+
+  const newProjects = projectSeeds.filter((t) => !existing.has(t.name));
+  if (newProjects.length === 0) {
+    return;
+  }
+
+  for (const { type, ...project } of newProjects) {
+    const insertQ = e.insert(e[`${type}Project`], {
+      ...project,
+      ...mapValues.fromList(
+        ['mouStart', 'mouEnd', 'estimatedSubmission'],
+        (key, { SKIP }) =>
```

Comment:
@CarsonF - Can you explain what is happening with this `SKIP` business?

Reply:
If the `project` doesn't specify that key (or rather it's falsy), the pair is omitted from the resulting object.
It would be similar to using `undefined` instead, and expecting code receiving that object to ignore that pair. Most things do, as does the QB code, so it's tomateo tomatto.